openpype/hosts/maya/plugins/publish/collect_render.py

```diff
@@ -69,388 +68,248 @@ class CollectMayaRender(pyblish.api.ContextPlugin):
-            options["renderGlobals"][key] = machine_list
+        # append full path
+        aov_dict = {}
+        default_render_folder = context.data.get('project_settings')\
+            .get('maya')\
+            .get('RenderSettings')\
+            .get('default_render_image_folder') or ""
+        # replace relative paths with absolute. Render products are
+        # returned as list of dictionaries.
+        publish_meta_path = None
+        for aov in exp_files:
+            full_paths = []
+            aov_first_key = list(aov.keys())[0]
+            for file in aov[aov_first_key]:
+                full_path = os.path.join(workspace, default_render_folder,
+                                         file)
+                full_path = full_path.replace("\\", "/")
+                full_paths.append(full_path)
+                publish_meta_path = os.path.dirname(full_path)
+            aov_dict[aov_first_key] = full_paths
+        full_exp_files = [aov_dict]
+        self.log.info(full_exp_files)
+
+        if publish_meta_path is None:
+            raise KnownPublishError("Unable to detect any expected output "
+                                    "images for: {}. Make sure you have a "
+                                    "renderable camera and a valid frame "
+                                    "range set for your renderlayer."
+                                    "".format(instance.name))
+
+        frame_start_render = int(self.get_render_attribute(
+            "startFrame", layer=layer_name))
+        frame_end_render = int(self.get_render_attribute(
+            "endFrame", layer=layer_name))
+
+        if (int(context.data['frameStartHandle']) == frame_start_render
+                and int(context.data['frameEndHandle']) == frame_end_render):  # noqa: W503, E501
+
+            handle_start = context.data['handleStart']
+            handle_end = context.data['handleEnd']
+            frame_start = context.data['frameStart']
+            frame_end = context.data['frameEnd']
+            frame_start_handle = context.data['frameStartHandle']
+            frame_end_handle = context.data['frameEndHandle']
+        else:
+            handle_start = 0
+            handle_end = 0
+            frame_start = frame_start_render
+            frame_end = frame_end_render
+            frame_start_handle = frame_start_render
+            frame_end_handle = frame_end_render
+
+        # find common path to store metadata
+        # so if image prefix is branching to many directories
+        # metadata file will be located in top-most common
+        # directory.
+        # TODO: use `os.path.commonpath()` after switch to Python 3
+        publish_meta_path = os.path.normpath(publish_meta_path)
+        common_publish_meta_path = os.path.splitdrive(
+            publish_meta_path)[0]
+        if common_publish_meta_path:
+            common_publish_meta_path += os.path.sep
+        for part in publish_meta_path.replace(
```

Comment:
hm I never trust `for` loops not expanding the `replace().split()` evaluation on every iteration so I'd be explicit and separate into a variable:
```
publish_path_parts = publish_meta_path.replace(common_publish_meta_path, "").split(os.path.sep)
for part in publish_path_parts:
```

Reply:
I believe Python shouldn't be doing that though. @iLLiCiTiT do we want to refactor this? I don't mind either.

Reply:
yeah it probably doesn't, just got used to avoid it because I got bit by this in the past, I forgot which language that was